lib/improver/blending/weighted_blend.py

```diff
@@ -515,14 +527,16 @@ def check_compatible_time_points(self, cube):
             ValueError : If blending over forecast reference time on a cube
                          with multiple times.
         """
-        if self.coord == "forecast_reference_time":
-            time_points = cube.coord("time").points
-            if len(time_points) > 1:
-                msg = ("For blending using the forecast_reference_time "
-                       "coordinate, the points within the time coordinate need"
-                       " to be the same. The time points within the input cube"
-                       " are {}".format(time_points))
-                raise ValueError(msg)
+        if self.timeblending is True:
+            return
+
+        time_points = cube.coord("time").points
+        if len(set(time_points)) > 1:
+            msg = ("For blending using the forecast_reference_time "
+                   "coordinate, the points within the time coordinate need"
+                   " to be the same. The time points within the input cube"
+                   " are {}".format(time_points))
```

Comment:
Could we update this error message to reflect the new logic?

Reply:
Done.